tensorrt_llm/serve/openai_disagg_server.py

```diff
@@ -487,6 +510,39 @@ async def send_completion_request(self, url: str, request: CompletionRequest) ->
     async def send_chat_request(self, url: str, request: ChatCompletionRequest) -> ChatCompletionResponse:
         return await self.send_request(url, request, "/v1/chat/completions", ChatCompletionResponse, self.create_chat_generator)
 
+    async def set_steady_clock_offsets(self, session: aiohttp.ClientSession):
+        STEADY_CLOCK_OFFSET_ENDPOINT = "/steady_clock_offset"
+        async def query_steady_clock_offset(server_url: str) -> tuple[Optional[float], Optional[float]]:
+            try:
+                originate_ts = get_steady_clock_now_in_seconds()
+                async with session.get(server_url + STEADY_CLOCK_OFFSET_ENDPOINT) as response:
+                    destination_ts = get_steady_clock_now_in_seconds()
+                    if response.status == 200:
+                        response = await response.json()
+                        # Compute the steady clock timestamp difference using the NTP clock synchronization algorithm. https://en.wikipedia.org/wiki/Network_Time_Protocol#Clock_synchronization_algorithm
+                        receive_ts = response['receive_ts']
+                        transmit_ts = response['transmit_ts']
+                        delay = (destination_ts - originate_ts) - (transmit_ts - receive_ts)
+                        offset = ((receive_ts - originate_ts) + (transmit_ts - destination_ts)) / 2
+                        return delay, offset
+                    else:
+                        return None, None
+            except Exception:
+                return None
+        async def set_steady_clock_offset(server_url: str, offset: float) -> Optional[float]:
```

Comment:
doesn't look like this function ever returns an `Optional[float]`. Should we change to `-> None`?